docs/pages/setup/reference/cli.mdx

```diff
@@ -1310,6 +1310,342 @@ $ tctl auth rotate --type=user --grace-period=200h
 $ tctl auth rotate --type=host --grace-period=8h
 ```
 
+### tctl sso test
+
+Perform end-to-end test of SSO flow using provided auth connector definition.
+
+The command supports all auth connector types: `github`, `oidc`, `saml`. The latter two require Teleport Enterprise.
+
+The testing consists of running a single end-to-end authentication request against provided auth connector definition.
+Once the request is finished, the results will be printed to console along with context-specific diagnostic informations.
+The test process will modify the list of configured auth connectors or result in certificates being issued.
+
+```code
+$ tctl [<global-flags>] sso test [<filename>]
+```
+
+<Admonition
+  type="note"
+  title="Important"
+>
+  To use this command, you must have access to `github_request`, `oidc_request`, `saml_request` resources (depending on the type of connector being tested).
+
+  If you receive "permission denied" error, ensure that have access to resources:
+
+```yaml
+- resources: [github_request]
+  verbs: [list,create,read,update,delete]
+- resources: [oidc_request]
+  verbs: [list,create,read,update,delete]
+- resources: [saml_request]
+  verbs: [list,create,read,update,delete]
+```
+</Admonition>
+
+
+#### Arguments
+
+- `[<filename>]` Connector resource definition file. Optional. Empty for stdin.
+
+#### Flags
+
+This command defines no flags.
+
+#### Global flags
+
+These flags are available for all commands `--debug, --config`. Run
+`tctl help <subcommand>` or see the [Global Flags section](#tctl-global-flags).
+
+#### Examples
+
+Test the auth connector from `connector.yaml`:
+
+```code
+$ tctl sso test connector.yaml
+```
+
+The command is designed to be used in conjunction with `tctl sso configure` family of commands:
+
+```code
+$ tctl sso configure ... | tctl sso test
+```
+
+The pipeline may also utilize `tee` to capture the connector generated with `tctl sso configure`. This will save the connector in `connector.yaml`:
+
+```code
+$ tctl sso configure ... | tee connector.yaml | tctl sso test
+```
+
+You can test an existing auth connector by combining the command with `tctl get`:
+
+```bsh
+$ tctl get saml/your-connector-name --with-secrets | tctl sso test
+````
+
+<Admonition
+  type="note"
+  title="Important"
+>
+  Make sure to include `--with-secrets` flag, or the exported auth connector will not be testable.
+</Admonition>
+
+### tctl sso configure github
+
+Configure GitHub auth connector.
+
+Required params `--id` and `--secret` come from the [Github OAuth app](https://docs.github.com/en/developers/apps/building-oauth-apps/creating-an-oauth-app).
+
+The flag `--teams-to-roles` can be provided multiple times to specify which Github Teams are assigned which roles.
```

Comment:
```suggestion
The flag `--teams-to-roles` can be provided multiple times to specify which Github Teams are assigned which roles.
In this example, members of the `octocats` team in GitHub will be assigned the `devs`, `access`, and `editor` roles in Teleport.
```

Reply:
Edit: I see you have this below. Either is fine, but it might make more sense here, as I found myself wondering what the command was doing as soon as I finished reading it.